x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/fulltext/Kql.java

```diff
@@ -26,7 +30,11 @@
  * Full text function that performs a {@link KqlQuery} .
  */
 public class Kql extends FullTextFunction {
-    public static final NamedWriteableRegistry.Entry ENTRY = new NamedWriteableRegistry.Entry(Expression.class, "Kql", Kql::new);
+    public static final NamedWriteableRegistry.Entry ENTRY = new NamedWriteableRegistry.Entry(Expression.class, "Kql", Kql::readFrom);
+
+    public Kql(Source source, Expression queryString, QueryBuilder queryBuilder) {
```

Comment:
I made them private and then the `EsqlNodeSubclassTests` started failing because AFAICT they expect that the argument number from the longest public constructor to correlate with the number of args from `NodeInfo`.
This to me seems to be a quirk of the `EsqlNodeSubclassTests` rather than a test catching a real bug with `FullTextFunction`s